pinot-core/src/main/java/com/linkedin/pinot/core/operator/docvaliterators/SingleValueIterator.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.operator.docvaliterators;
+
+import com.linkedin.pinot.core.common.BlockSingleValIterator;
+import com.linkedin.pinot.core.io.reader.ReaderContext;
+import com.linkedin.pinot.core.io.reader.SingleColumnSingleValueReader;
+
+
+public class SingleValueIterator extends BlockSingleValIterator {
+  private SingleColumnSingleValueReader<? super ReaderContext> _reader;
+  private final int _numDocs;
+  private final ReaderContext _context;
+  private int _currentDocId;
+
+  public SingleValueIterator(SingleColumnSingleValueReader<? super ReaderContext> reader, int numDocs) {
+    _reader = reader;
+    _numDocs = numDocs;
+    _context = _reader.createContext();
+    _currentDocId = 0;
+  }
+
+  @Override
+  public int nextIntVal() {
+    return _reader.getInt(_currentDocId++, _context);
+  }
+
+  @Override
+  public long nextLongVal() {
+    return _reader.getLong(_currentDocId++, _context);
+  }
+
+  @Override
+  public float nextFloatVal() {
+    return _reader.getFloat(_currentDocId++, _context);
+  }
+
+  @Override
+  public double nextDoubleVal() {
+    return _reader.getDouble(_currentDocId++, _context);
+  }
+
+  @Override
+  public String nextStringVal() {
+    return _reader.getString(_currentDocId++, _context);
+  }
+
+  @Override
+  public boolean hasNext() {
+    return _currentDocId < _numDocs;
```

Comment:
Seems either the variable _currentDocId is named incorrectly or the logic is incorrect. For example if _numDocs = 10, and _currentDocId = 9, it will return true. If that is desired, that means the variable name is confusing.